README.md

```diff
@@ -97,6 +97,106 @@ The evaluation (NER and entity linking) can be run in the [evaluation/](evaluati
+
+### 1. `Corpus`
+The `Corpus` class represents a collection of documents with annotations. Each document in the corpus must adhere to the following format:
+```json
+{
+    "doc_id": "unique_document_identifier",
+    "text": "document_text",
+    "annotations": [
+        {"label": "LABEL", "start_span": "start_position", "end_span": "end_position"},
+        ...
+    ]
+}
+```
+
+#### Methods:
+- `__init__(data: list)`: Initializes the corpus. The `data` must be a list of documents with the structure mentioned above.
+- `__len__()`: Returns the number of documents in the corpus.
+- `get_entities()`: Returns a set of all unique entity labels in the corpus.
+- `split(split: float)`: Splits the corpus into two sets (training and testing) based on the provided split ratio.
+
+### 2. `CorpusPreProcessor`
+The `CorpusPreProcessor` class is responsible for applying transformations and filters to the corpus.
+
+#### Methods:
+- `__init__(corpus: Corpus)`: Initializes the preprocessor with a `Corpus` object.
+- `filter_labels(labels: list)`: Filters the annotations to keep only those with labels that match the provided list.
+- `merge_annotations()`: Merges overlapping or adjacent annotations based on their span. If a collision is found, the annotations are merged.
+- `split_data(test_split_percentage: float)`: Splits the corpus into training and test sets based on the provided split percentage.
+
+### 4. `CorpusTokenizer`
+The `CorpusTokenizer` class tokenizes the corpus using a given tokenizer (e.g., from the Hugging Face library) and prepares it for training by splitting it into context windows.
+
+#### Parameters:
+- `corpus: CorpusPreProcessor`: The preprocessed corpus to tokenize.
+- `tokenizer`: A tokenizer that tokenizes the text (e.g., Hugging Face's tokenizers).
+- `context_size`: The size of the context to be added before and after the main token sequence (optional).
+
+#### Methods:
+- `__tokenize()`: Tokenizes the entire corpus.
+- `__split()`: Splits the tokenized corpus into windows, handling context and special tokens (if any).
+
+### 5. `CorpusDataset`
+This class wraps the tokenized corpus into a dataset compatible with PyTorch's `DataLoader`.
+
+#### Parameters:
+- `tokenized_corpus: CorpusTokenizer`: The tokenized corpus.
+- `transforms`: Optional transformations to apply to each sample.
+- `augmentations`: Optional augmentations to apply to each sample.
+
+
+
+## Usage Example
+
+```python
+# First, create a generic Corpus
+spanishCorpus = Spanish_Biomedical_NER_Corpus(
+    "../dataset/merged_data_subtask1_train.tsv", 
+    "../dataset/documents"
+)
+
+# Create a Corpus PreProcessor, which handles certain preprocessing tasks: 
+# merging annotations, filtering labels, and splitting the data.
+spanishCorpusProcessor = CorpusPreProcessor(spanishCorpus)
+spanishCorpusProcessor.merge_annoatation()
```

Comment:
This function is optional? It does not have arguments, how the user knows if he should run or not?

If it is something that we should always run, then it should not be a public function. Instead, we should just run it inside, even if it does nothing for some datasets.

Reply:
The function is optional. It merges overlapping annotaitons, it does not need any arguments.  

Not sure if the remainder code will support overlapping entities, but the function is not mandatory to run. Highly advised to run it.